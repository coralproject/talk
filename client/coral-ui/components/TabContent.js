import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './TabContent.css';

function getRootClassName(className) {
  return cn('talk-tab-content', className, styles.root);
}

/**
 * The `TabContent` component accepts `TabPane` components to render
 * the content of a `Tab`.
 */
const TabContent = ({ children, className, activeTab, sub, ...rest }) => (
  <div {...rest} className={getRootClassName(className)}>
    {React.Children.toArray(children)
      .filter(child => child.props.tabId === activeTab)
      .map((child, i) =>
        React.cloneElement(child, {
          tabId: child.props.tabId !== undefined ? child.props.tabId : i,
          sub,
        })
      )}
  </div>
);

TabContent.propTypes = {
  // className to be added to the root element.
  className: PropTypes.string,

  // activeTab should be set to the currently active tabId.
  activeTab: PropTypes.string,

  // Sub indicates that this component belongs to a sub-tab-panel.
  sub: PropTypes.bool,
};

export default TabContent;
