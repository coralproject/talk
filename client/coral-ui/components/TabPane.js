import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

function getRootClassName(className) {
  return cn('talk-pane', className);
}

/**
 * The `TabPane` component is used inside the `TabContent` component to render
 * the content of a `Tab`.
 */
const TabPane = ({ children, className, tabId: _a, sub: _b, ...rest }) => (
  <div {...rest} className={getRootClassName(className)}>
    {children}
  </div>
);

TabPane.propTypes = {
  // className to be added to the root element.
  className: PropTypes.string,

  tabId: PropTypes.string,

  // Sub indicates that this component belongs to a sub-tab-panel.
  // This is injected by the `TabContent` component.
  sub: PropTypes.bool,
};

export default TabPane;
