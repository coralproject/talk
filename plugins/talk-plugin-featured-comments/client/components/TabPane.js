import React from 'react';
import Comment from '../containers/Comment';
import LoadMore from './LoadMore';
import { getErrorMessages } from 'plugin-api/beta/client/utils';

class TabPane extends React.Component {
  state = {
    loadingState: '',
  };

  loadMore = () => {
    this.setState({ loadingState: 'loading' });
    this.props
      .loadMore()
      .then(() => {
        this.setState({ loadingState: 'success' });
      })
      .catch(error => {
        this.setState({ loadingState: 'error' });
        this.props.notify('error', getErrorMessages(error));
      });
  };

  render() {
    const {
      root,
      data,
      asset: { featuredComments, ...asset },
      viewComment,
    } = this.props;
    return (
      <div>
        {featuredComments.nodes.map(comment => (
          <Comment
            key={comment.id}
            root={root}
            data={data}
            comment={comment}
            asset={asset}
            viewComment={viewComment}
          />
        ))}
        {featuredComments.hasNextPage && (
          <LoadMore
            loadMore={this.loadMore}
            loadingState={this.state.loadingState}
          />
        )}
      </div>
    );
  }
}

export default TabPane;
