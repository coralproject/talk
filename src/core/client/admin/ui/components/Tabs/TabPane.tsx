import React, { FunctionComponent } from "react";

export interface TabBarProps {
  /**
   * Convenient prop to override the root styling.
   */
  className?: string;
  /**
   * Name of the tab
   */
  tabID: string;
}

const TabPane: FunctionComponent<TabBarProps> = props => {
  const { className, children, tabID, ...rest } = props;
  return (
    <section
      {...rest}
      className={className}
      key={tabID}
      id={`tabPane-${tabID}`}
      role="tabpanel"
      aria-labelledby={`tab-${tabID}`}
    >
      {children}
    </section>
  );
};

export default TabPane;
