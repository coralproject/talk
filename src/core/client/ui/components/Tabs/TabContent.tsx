import React, { StatelessComponent } from "react";

export interface TabContentProps {
  /**
   * Active tab id/name
   */
  activeTab?: string;
}

const TabContent: StatelessComponent<TabContentProps> = props => {
  const { children, activeTab } = props;
  return (
    <>
      {React.Children.toArray(children)
        .filter(
          (child: React.ReactElement<any>) => child.props.tabId === activeTab
        )
        .map((child: React.ReactElement<any>, i) =>
          React.cloneElement(child, {
            tabId: child.props.tabId ? child.props.tabId : i,
          })
        )}
    </>
  );
};

export default TabContent;
