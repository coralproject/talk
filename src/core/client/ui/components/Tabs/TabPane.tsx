import React, { StatelessComponent } from "react";

export interface TabBarProps {
  /**
   * Convenient prop to override the root styling.
   */
  className?: string;
  /**
   * Name of the tab
   */
  tabId: string;
}

const TabPane: StatelessComponent<TabBarProps> = props => {
  const { className, children, tabId } = props;
  return (
    <section
      className={className}
      key={tabId}
      id={`tabPane-${tabId}`}
      role="tabpanel"
      aria-labelledby={`tab-${tabId}`}
    >
      {children}
    </section>
  );
};

export default TabPane;
