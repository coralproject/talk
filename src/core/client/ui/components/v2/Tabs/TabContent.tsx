import cn from "classnames";
import React, { FunctionComponent } from "react";

export interface TabContentProps {
  /**
   * Active tab id/name
   */
  activeTab?: string;
  /**
   * classNames
   */
  className?: string;
}

const TabContent: FunctionComponent<TabContentProps> = props => {
  const { children, activeTab, className } = props;
  return (
    <>
      {React.Children.toArray(children)
        .filter(
          (child: React.ReactElement<any>) => child.props.tabID === activeTab
        )
        .map((child: React.ReactElement<any>, i) =>
          React.cloneElement(child, {
            tabID: child.props.tabID ? child.props.tabID : i,
            className: cn(className, child.props.className),
          })
        )}
    </>
  );
};

export default TabContent;
