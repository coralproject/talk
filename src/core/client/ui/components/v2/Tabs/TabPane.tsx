import React, { FunctionComponent } from "react";

export interface TabBarProps {
  /**
   * Convenient prop to override the root styling.
   */
  className?: string;
  /**
   * Name of the tab
   */
  tabID: string;

  "aria-labelledBy"?: string;

  children?: React.ReactNode;
}

const TabPane: FunctionComponent<TabBarProps> = (props) => {
  const {
    className,
    children,
    tabID,
    "aria-labelledBy": ariaLabelledBy,
    ...rest
  } = props;
  return (
    <section
      {...rest}
      className={className}
      key={tabID}
      id={`tabPane-${tabID}`}
      aria-labelledby={ariaLabelledBy || `tab-${tabID}`}
    >
      {children}
    </section>
  );
};

export default TabPane;
