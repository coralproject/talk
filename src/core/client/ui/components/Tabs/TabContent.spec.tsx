import React from "react";
import TestRenderer from "react-test-renderer";

import TabContent from "./TabContent";
import TabPane from "./TabPane";

it("renders correctly", () => {
  const renderer = TestRenderer.create(
    <TabContent activeTab="one">
      <TabPane tabId="one">Hola One</TabPane>
      <TabPane tabId="two">Hola Two</TabPane>
      <TabPane tabId="three">Hola Three</TabPane>
    </TabContent>
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});

it("sets initial tab as active, renders only one", () => {
  const renderer = TestRenderer.create(
    <TabContent activeTab="one">
      <TabPane tabId="one">Hola One</TabPane>
      <TabPane tabId="two">Hola Two</TabPane>
      <TabPane tabId="three">Hola Three</TabPane>
    </TabContent>
  );

  const testInstance = renderer.root;
  expect(testInstance.findByType(TabContent).props.activeTab).toBe("one");
  const panes = testInstance.findAllByType(TabPane);
  expect(panes.length).toBe(1);
});
