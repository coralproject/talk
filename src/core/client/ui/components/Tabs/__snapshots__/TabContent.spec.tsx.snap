// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  aria-labelledby="tab-one"
  id="tabPane-one"
  role="tabpanel"
>
  Hola One
</section>
`;
